cmd/snap-seccomp/main_test.go

```diff
@@ -0,0 +1,404 @@
+		c.Check(err, ErrorMatches, t.errMsg, Commentf("%q errors in unexpected ways, got: %q expected %q", t.inp, err, t.errMsg))
+	}
+}
+
+// ported from test_restrictions_working_args_socket
+func (s *snapSeccompSuite) TestRestrictionsWorkingArgsSocket(c *C) {
+	bpf.VmEndianness = nativeEndian()
+
+	for _, pre := range []string{"AF", "PF"} {
+		for _, i := range []string{"UNIX", "LOCAL", "INET", "INET6", "IPX", "NETLINK", "X25", "AX25", "ATMPVC", "APPLETALK", "PACKET", "ALG", "CAN", "BRIDGE", "NETROM", "ROSE", "NETBEUI", "SECURITY", "KEY", "ASH", "ECONET", "SNA", "IRDA", "PPPOX", "WANPIPE", "BLUETOOTH", "RDS", "LLC", "TIPC", "IUCV", "RXRPC", "ISDN", "PHONET", "IEEE802154", "CAIF", "NFC", "VSOCK", "MPLS", "IB"} {
+			seccompWhitelist := fmt.Sprintf("socket %s_%s", pre, i)
+			bpfInputGood := fmt.Sprintf("socket;native;%s_%s", pre, i)
+			bpfInputBad := "socket;native;99999"
+			simulateBpf(c, seccompWhitelist, bpfInputGood, main.SeccompRetAllow)
+			simulateBpf(c, seccompWhitelist, bpfInputBad, main.SeccompRetKill)
+
+			for _, j := range []string{"SOCK_STREAM", "SOCK_DGRAM", "SOCK_SEQPACKET", "SOCK_RAW", "SOCK_RDM", "SOCK_PACKET"} {
+				seccompWhitelist := fmt.Sprintf("socket %s_%s %s", pre, i, j)
+				bpfInputGood := fmt.Sprintf("socket;native;%s_%s,%s", pre, i, j)
+				bpfInputBad := fmt.Sprintf("socket;native;%s_%s,9999", pre, i)
+				simulateBpf(c, seccompWhitelist, bpfInputGood, main.SeccompRetAllow)
+				simulateBpf(c, seccompWhitelist, bpfInputBad, main.SeccompRetKill)
+			}
+		}
+	}
+
+	for _, i := range []string{"NETLINK_ROUTE", "NETLINK_USERSOCK", "NETLINK_FIREWALL", "NETLINK_SOCK_DIAG", "NETLINK_NFLOG", "NETLINK_XFRM", "NETLINK_SELINUX", "NETLINK_ISCSI", "NETLINK_AUDIT", "NETLINK_FIB_LOOKUP", "NETLINK_CONNECTOR", "NETLINK_NETFILTER", "NETLINK_IP6_FW", "NETLINK_DNRTMSG", "NETLINK_KOBJECT_UEVENT", "NETLINK_GENERIC", "NETLINK_SCSITRANSPORT", "NETLINK_ECRYPTFS", "NETLINK_RDMA", "NETLINK_CRYPTO", "NETLINK_INET_DIAG"} {
+		for _, j := range []string{"AF_NETLINK", "PF_NETLINK"} {
+			seccompWhitelist := fmt.Sprintf("socket %s - %s", j, i)
+			bpfInputGood := fmt.Sprintf("socket;native;%s,0,%s", j, i)
+			bpfInputBad := fmt.Sprintf("socket;native;%s,0,99", j)
+			simulateBpf(c, seccompWhitelist, bpfInputGood, main.SeccompRetAllow)
+			simulateBpf(c, seccompWhitelist, bpfInputBad, main.SeccompRetKill)
+		}
+	}
+}
+
+// ported from test_restrictions_working_args_quotactl
+func (s *snapSeccompSuite) TestRestrictionsWorkingArgsQuotactl(c *C) {
+	bpf.VmEndianness = nativeEndian()
+
+	for _, arg := range []string{"Q_QUOTAON", "Q_QUOTAOFF", "Q_GETQUOTA", "Q_SETQUOTA", "Q_GETINFO", "Q_SETINFO", "Q_GETFMT"} {
+		seccompWhitelist := fmt.Sprintf("quotactl %s", arg)
+		bpfInputGood := fmt.Sprintf("quotactl;native;%s", arg)
+		simulateBpf(c, seccompWhitelist, bpfInputGood, main.SeccompRetAllow)
+		for _, bad := range []string{"quotactl;native;99999", "read;native;"} {
+			simulateBpf(c, seccompWhitelist, bad, main.SeccompRetKill)
+		}
```

Comment:
You are missing Q_SYNC, Q_XQUOTAON, Q_XQUOTAOFF, Q_XGETQUOTA, Q_XSETQLIM, Q_XGETQSTAT and Q_XQUOTARM.